From f8305cdfff78197518c2b3bfd95963d997e8a5fd Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Tue, 19 Jul 2016 16:49:06 +0200
Subject: [PATCH 1/2] ssm: Close file descriptors on popen

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 ssmlib/misc.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ssmlib/misc.py b/ssmlib/misc.py
index dce6ce6..9cbb1d1 100644
--- a/ssmlib/misc.py
+++ b/ssmlib/misc.py
@@ -474,7 +474,7 @@ def run(cmd, show_cmd=False, stdout=False, stderr=True, can_fail=False,
             cmd[i] = str(item)
 
     proc = subprocess.Popen(cmd, stdout=stdout,
-                            stderr=stderr, stdin=stdin)
+                            stderr=stderr, stdin=stdin, close_fds=True)
 
     if stdin_data is not None:
 
-- 
1.8.3.1

