From 73462b33f8c293252d7bb920383a59884bd096de Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Fri, 22 Jun 2018 14:37:15 +0200
Subject: [PATCH 3/5] tests: Test ssm list with exported volume groups

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 tests/bashtests/011-lvm-list.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tests/bashtests/011-lvm-list.sh b/tests/bashtests/011-lvm-list.sh
index e9eb80a..80d3ea0 100755
--- a/tests/bashtests/011-lvm-list.sh
+++ b/tests/bashtests/011-lvm-list.sh
@@ -327,5 +327,14 @@ for fs in $TEST_FS; do
 done
 ssm -f remove $SSM_LVM_DEFAULT_POOL
 
+# Check how ssm handles exported volume groups
+# This was originally fixed by commit
+# 7449db6 ("ssm: add a workaround for lvm and exported volumes")
+ssm add $TEST_DEVS
+vgexport $SSM_LVM_DEFAULT_POOL
+ssm list
+vgimport $SSM_LVM_DEFAULT_POOL
+ssm -f remove $SSM_LVM_DEFAULT_POOL
+
 # Some situation should fail
 not ssm list wrong_type
-- 
2.17.1

